src/Build/Definition/ProjectCollection.cs

```diff
@@ -1041,11 +1056,12 @@ public Project LoadProject(string fileName, string toolsVersion)
         /// <returns>A loaded project.</returns>
         public Project LoadProject(string fileName, IDictionary<string, string> globalProperties, string toolsVersion)
         {
+            ErrorUtilities.VerifyThrowArgumentLength(fileName, nameof(fileName));
+            fileName = FileUtilities.NormalizePath(fileName);
+
+            var buildEventContext = new BuildEventContext(0 /* node ID */, BuildEventContext.InvalidTargetId, BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTaskId);
```

Comment:
Looks like this is only used in the exceptional case. Consider only allocating in the `catch` block.

Reply:
nice catch